### ISSUE TITLE
npm 3.x installing global packages in program files\node\node_modules

### ISSUE THREAD

**Issue by robertbaker** [npm/npm#8980]
npm-windows-upgrade claims it just uses the official npm installer.

Clean remove node. (delete `%programfiles%\node` and `%appdata%\npm` and `%appdata\npm-cache` Install Node 4.x. Install `npm-windows-upgrade`, update to latest npm 3.x. Install a global package. They are stored in Program Files\nodejs\node_modules.

In NPM 2, global packages were installed in %appdata%\npm.

**Comment by cchamberlain** [comment]
Sounds like it's using the nodejs msi installation for installing that.  When I install on Windows I set environment variable NPM_CONFIG_PREFIX to %USERPROFILE%\local\npm prior to installation and the installer will then install global modules into that directory.  Npm windows upgrade will ask npm for its installed location and will default to the program files location if it can't find anything.

**Comment by robertbaker** [comment]
npm packages should be installed in %localappdata%\npm, previously it used %appdata%\npm, but that wasn't proper either.

**Comment by ehsalazar** [comment]
@robertbaker it sounds like you were having environment issues. Is this still a problem you're experiencing? If so, these might help: 
- [Troubleshooting - Upgrading on Windows](https://github.com/npm/npm/wiki/Troubleshooting#upgrading-on-windows)
- [npm-windows-upgrade](https://github.com/felixrieseberg/npm-windows-upgrade)

If this has resolved, please let us know and we'll close out this issue.

**Comment by robertbaker** [comment]
@ehsalazar not exactly. It seems that NPM v3 stores global packages in the node program directory.

In NPM 2, global packages are stored in %appdata%\npm.

NPM 2, you didn't need to elevate cmd to install globals, the downside to that approach was storing them in appdata/roaming (not appdata\local), which caused domain environment login performance issues,

I use npm-windows-upgrade to update to the latest 3.x releases. I believe storing global packages in the Node folder is by design now, which I'm personally fine with.

If NPM 3 global packages should still be installed in %appdata%\npm, then this issue is not resolved, and a bug would need reported to npm-windows-upgrade

**Comment by othiym23** [comment]
@robertbaker In your original writeup, I don't see any mention of removing your `npmrc` -- did you clear that out as well when you uninstalled npm and Node? In general, Node + `npm@3` shouldn't install to `Program Files\nodejs`, and if that were happening in general, it would be a serious issue. We've gotten only sporadic reports of this happening, and so far eventually all of those have come down to configuration issues. Can you run `npm config ls -l` and put the output up on gist.github.com (be sure to review it and XXX out anything that looks sensitive)?

**Comment by robertbaker** [comment]
I didn't remove npmrc.

npm-windows-upgrade claims it just uses the official npm installer.

Clean remove node. Install Node. Install npm-windows-upgrade, update to latest 3.x. Install a global package. They are stored in Program Files\nodejs.

It's been that way for months, honestly is fine with me. Storing global packages in %appdata%\npm is a bad idea, should be stored in %localappdata%\npm then.

**Comment by robertbaker** [comment]
```
C:\Users\rober>npm config ls -l
; cli configs
long = true
user-agent = "npm/2.14.4 node/v4.1.2 win32 x64"

; builtin config undefined
prefix = "C:\\Users\\rober\\AppData\\Roaming\\npm"

; default values
access = null
always-auth = false
bin-links = true
browser = null
ca = null
cache = "C:\\Users\\rober\\AppData\\Roaming\\npm-cache"
cache-lock-retries = 10
cache-lock-stale = 60000
cache-lock-wait = 10000
cache-max = null
cache-min = 10
cafile = undefined
cert = null
color = true
depth = null
description = true
dev = false
editor = "notepad.exe"
engine-strict = false
fetch-retries = 2
fetch-retry-factor = 10
fetch-retry-maxtimeout = 60000
fetch-retry-mintimeout = 10000
force = false
git = "git"
git-tag-version = true
global = false
globalconfig = "C:\\Users\\rober\\AppData\\Roaming\\npm\\etc\\npmrc"
globalignorefile = "C:\\Users\\rober\\AppData\\Roaming\\npm\\etc\\npmignore"
group = 0
heading = "npm"
https-proxy = null
if-present = false
ignore-scripts = false
init-author-email = ""
init-author-name = ""
init-author-url = ""
init-license = "ISC"
init-module = "C:\\Users\\rober\\.npm-init.js"
init-version = "1.0.0"
json = false
key = null
link = false
local-address = undefined
loglevel = "warn"
; long = false (overridden)
message = "%s"
node-version = "4.1.2"
npat = false
onload-script = null
optional = true
parseable = false
; prefix = "C:\\Program Files\\nodejs" (overridden)
production = false
proprietary-attribs = true
proxy = null
rebuild-bundle = true
registry = "https://registry.npmjs.org/"
rollback = true
save = false
save-bundle = false
save-dev = false
save-exact = false
save-optional = false
save-prefix = "^"
scope = ""
searchexclude = null
searchopts = ""
searchsort = "name"
shell = "C:\\WINDOWS\\system32\\cmd.exe"
shrinkwrap = true
sign-git-tag = false
spin = true
strict-ssl = true
tag = "latest"
tag-version-prefix = "v"
tmp = "C:\\Users\\rober\\AppData\\Local\\Temp"
umask = 0
unicode = true
unsafe-perm = true
usage = false
user = 0
; user-agent = "npm/{npm-version} node/{node-version} {platform} {arch}" (overridden)
userconfig = "C:\\Users\\rober\\.npmrc"
version = false
versions = false
viewer = "browser"
```

**Comment by robertbaker** [comment]
I just clean removed nodejs and npm and installed node 5.0, which includes npm 3.3.6. I ran `npm i -g npm-windows-upgrade` which installed the global package in `%appdata%\npm\node_modules`. I updated npm using `npm-windows-upgrade` and re-installed the global package. It put `npm-windows-upgrade` into `%programfiles%\nodejs\node_modules`.

The upgrade tool does not generate the needed npmrc file.

Personally I don't understand the need for the npmrc file unless you are a non-admin user.

Regardless I think storing the global packages in %appdata% is a mistake, it causes issues with roaming profiles. It would be better to store them in `%localappdata%` instead

I filed an issue with `npm-windows-upgrade` to address the missing npmrc file. You guys should chime in. @iarna @othiym23 
https://github.com/felixrieseberg/npm-windows-upgrade/issues/46
